src/authentication/json-api-authorization-spec.ts

```diff
@@ -319,18 +319,19 @@ describe('authorization for api routes', () => {
         expect(api.deleteCommentHandler).to.not.have.been.called;
         expect(response.statusCode).to.eq(401);
       });
-      it('should allow deleting a comment for an open deployment without authentication', async () => {
+      it('should not allow deleting a comment for an open deployment without authentication', async () => {
         // Arrange
         const { server, authentication, api } = await arrangeCommentRemoval(true, true);
         // Act
-        await server.inject({
+        const response = await server.inject({
           method: 'DELETE',
           url: 'http://foo.com/comments/1',
         });
         // Assert
         expect(api.getComment).to.have.been.calledOnce;
-        expect(authentication.isOpenDeployment).to.have.been.calledOnce;
-        expect(api.deleteCommentHandler).to.have.been.calledOnce;
+        expect(authentication.isOpenDeployment).to.not.have.been.called;
```

Comment:
These tests do follow the implementations quite closely. In general I've included the knowledge whether the stubbed methods should have been called or not as part of the tests. 

It might be a good idea to only test the response status code in the negative cases, but I'd like to be consistent and do that for all these tests in a separate PR when deemed appropriate.